File: gitdot-web/packages/api/src/endpoint/auth/account/verify-user-email.ts
import { z } from "zod";
import { UserEmailResource } from "../../../resource";
import type { Endpoint } from "../../endpoint";

export const VerifyUserEmailRequest = z.object({
  email: z.string(),
  code: z.string(),
});
export type VerifyUserEmailRequest = z.infer<typeof VerifyUserEmailRequest>;

export const VerifyUserEmailResponse = UserEmailResource;
export type VerifyUserEmailResponse = z.infer<typeof VerifyUserEmailResponse>;

export const VerifyUserEmail = {
  path: "/auth/account/verify-email",
  method: "POST",
  request: VerifyUserEmailRequest,
  response: VerifyUserEmailResponse,
} as const satisfies Endpoint;
export type VerifyUserEmail = typeof VerifyUserEmail;

File: gitdot-api/src/endpoint/auth/email/verify_user_email.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::user::UserEmailResource};

pub struct VerifyUserEmail;

impl Endpoint for VerifyUserEmail {
    const PATH: &'static str = "/auth/email/verify-email";
    const METHOD: http::Method = http::Method::POST;

    type Request = VerifyUserEmailRequest;
    type Response = VerifyUserEmailResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct VerifyUserEmailRequest {
    pub email: String,
    pub code: String,
}

pub type VerifyUserEmailResponse = UserEmailResource;
